homeassistant/components/niu/__init__.py

```diff
@@ -0,0 +1,63 @@
+"""The Niu integration."""
+import asyncio
+import logging
+
+from niu import NiuCloud
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+# Want to support the sensor platform
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Niu component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Niu from a config entry."""
+    # Get data from config
+    config = entry.data
+    username = entry.title
+    password = config[CONF_PASSWORD]
+
+    # First have to define the domain
+    hass.data[DOMAIN] = {}
```

Comment:
What if the user sets up multiple instances of the integration?
The entry setup of the second instance would delete the first instance now.

```suggestion
    hass.data.setdefault(DOMAIN, {})
```